### PR TITLE
Add QNN glossary entry

### PR DESCRIPTION
**Pull request by mariaschuld**
**Title:**

 Adding a new entry on quantum neural nets.

**Possible Drawbacks:**

I am unsure if  I should add more references, since the style is very formal? Or shorten and make it less technical?

### CODE REVIEW

File: glossary.rst [hunk 1]
```diff
@@ -54,8 +58,9 @@ Key Concepts
 
     /glossary/automatic_differentiation
     /glossary/circuit_ansatz
```
Comment (mariaschuld) [inline glossary.rst:60]: Just ordering alphabetically here...

**Comment by josh146** [comment]
<h3>Website build</h3>
<strong>Commit:</strong> 9db55ed9e6bbb13d69f4f9319924c73d2521b219

<strong>Circle build number:</strong> 1033

<strong>Built <code>demos</code> directory:</strong> <a href=https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_demos.zip>https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_demos.zip</a>

<strong>Backreferences zip:</strong> <a href=https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_backreferences.zip>https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_backreferences.zip</a>

<strong>Website zip:</strong> <a href=https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_html.zip>https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_html.zip</a>

<strong>Website url:</strong> <a href=https://1033-214003948-gh.circle-artifacts.com/0/_build/html/index.html>https://1033-214003948-gh.circle-artifacts.com/0/_build/html/index.html</a>

<em>Please double check the rendered website build to make sure everything is correct.</em>

File: glossary/quantum_neural_network.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Quantum versions of feed-forward neural networks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: ../_static/concepts/qnn1.png
+    :align: center
+    :width: 50%
+    :target: javascript:void(0);
+
+Already in the 1990s, quantum physicists have tried to come up with "quantum" versions of recurrent and feed-forward neural networks `Schuld et al. (2014) <https://arxiv.org/abs/1408.7005>`_. The models were attempts to recover the modular structure as well as the nonlinear activation function of neural networks in the language of quantum algorithms. However, chains of linear and nonlinear computations are rather "unnatural" for quantum computers. [#]_
```
Comment (mariaschuld) [inline glossary/quantum_neural_network.rst:21]:
```suggestion
`Already in the 1990s, <https://arxiv.org/abs/1408.7005>`_ quantum physicists have tried to come up with "quantum versions" of recurrent and feed-forward neural networks. The models were attempts to translate the modular structure as well as the nonlinear activation functions of neural networks into the language of quantum algorithms. However, one could argue that chains of linear and nonlinear computations are rather "unnatural" for quantum computers [#]_. 
```

File: glossary/quantum_neural_network.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: ../_static/concepts/qnn1.png
+    :align: center
+    :width: 50%
+    :target: javascript:void(0);
+
+`Already in the 1990s, <https://arxiv.org/abs/1408.7005>`_ quantum physicists have tried to come up with "quantum versions" of recurrent and feed-forward neural networks. The models were attempts to translate the modular structure as well as the nonlinear activation functions of neural networks into the language of quantum algorithms. However, one could argue that chains of linear and nonlinear computations are rather "unnatural" for quantum computers [#]_. 
+
+Also more recent research has tried to tackle this problem, and suggested special measurement schemes, or modifications of the neural nets that make them more ameneable to quantum computing, but the advantage of these models for machine learning is still not conclusively established.
```
Comment (mariaschuld) [inline glossary/quantum_neural_network.rst:23]:
```suggestion
More recent research has tackled this problem, suggesting special measurement schemes or modifications of the neural nets that make them more amenable to quantum computing, but the advantage of these models for machine learning is still not conclusively established.
```

File: glossary/quantum_neural_network.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+Boltzmann machines, which are probabilistic graphical models that can be understood as stochastic recurrent neural networks, play an important role in the quantum machine learning literature. For example, it was suggested to use samples from a quantum computer to train classical Boltzmann machines, or to interpret spins as physical units of a "quantum" Boltzmann machine model.
+
+
+Variational circuits 
+~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: ../_static/concepts/qnn2.png
+    :align: center
+    :width: 50%
```
Comment (mariaschuld) [inline glossary/quantum_neural_network.rst:36]:
```suggestion
    :width: 60%
```

File: glossary/quantum_neural_network.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+A quantum neural network (QNN) is a machine learning model or algorithm that combines concepts from **quantum computing** and **artifical neural networks**. 
+
+Over the past decades, the term has been used to describe a variety of ideas, ranging from quantum computers *emulating* the exact computations of neural nets, to general *trainable* quantum circuits that bear only little resemblence with the multi-layer perceptron structure.
+
+Quantum versions of feed-forward neural networks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: ../_static/concepts/qnn1.png
+    :align: center
+    :width: 50%
```
Comment (mariaschuld) [inline glossary/quantum_neural_network.rst:18]:
```suggestion
    :width: 60%
```

File: glossary/quantum_neural_network.rst [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+Variational circuits 
+~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: ../_static/concepts/qnn2.png
+    :align: center
+    :width: 60%
+    :target: javascript:void(0);
+
+Increasingly, the term "quantum neural network" is used in a much looser analogy to refer to variational or parametrized quantum circuits. While mathematically rather different from the inner workings of neural networks, the name highlights  the "modular" nature of quantum gates in a circuit, as well as the wide use of tricks from training neural networks used in the optimization of quantum algorithms.
```
Comment (mariaschuld) [inline glossary/quantum_neural_network.rst:39]:
```suggestion
Increasingly, the term "quantum neural network" is used to refer to variational or parametrized quantum circuits. While mathematically rather different from the inner workings of neural networks, the analogy highlights  the "modular" nature of quantum gates in a circuit, as well as the wide use of tricks from training neural networks used in the optimization of quantum algorithms. 
```

**Comment by mariaschuld** [comment]
> ### Website build
> **Commit:** [9db55ed](https://github.com/XanaduAI/qml/commit/9db55ed9e6bbb13d69f4f9319924c73d2521b219)
> **Circle build number:** 1033
> 
> **Built `demos` directory:** https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_demos.zip
> 
> **Backreferences zip:** https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_backreferences.zip
> 
> **Website zip:** https://1033-214003948-gh.circle-artifacts.com/0/tmp/qml_html.zip
> 
> **Website url:** https://1033-214003948-gh.circle-artifacts.com/0/_build/html/index.html
> 
> _Please double check the rendered website build to make sure everything is correct._

This may be slightly outdated...

File: glossary/quantum_neural_network.rst [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+`Already in the 1990s, <https://arxiv.org/abs/1408.7005>`_ quantum physicists have tried to come up with "quantum versions" of recurrent and feed-forward neural networks. The models were attempts to translate the modular structure as well as the nonlinear activation functions of neural networks into the language of quantum algorithms. However, one could argue that chains of linear and nonlinear computations are rather "unnatural" for quantum computers [#]_. 
+
+More recent research has tackled this problem, suggesting special measurement schemes or modifications of the neural nets that make them more amenable to quantum computing, but the advantage of these models for machine learning is still not conclusively established.
+
+Quantum versions of Boltzmann machines
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Boltzmann machines, which are probabilistic graphical models that can be understood as stochastic recurrent neural networks, play an important role in the quantum machine learning literature. For example, it was suggested to use samples from a quantum computer to train classical Boltzmann machines, or to interpret spins as physical units of a "quantum" Boltzmann machine model.
```
[the remark this answers is not in the document]
Comment (thisac) [reply glossary/quantum_neural_network.rst:28]: Also, might it be nice to expand with further links to articles where these "quantum" Boltzmann machines are used?

File: glossary/quantum_neural_network.rst [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+Quantum versions of feed-forward neural networks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: ../_static/concepts/qnn1.png
+    :align: center
+    :width: 60%
+    :target: javascript:void(0);
+
+`Already in the 1990s, <https://arxiv.org/abs/1408.7005>`_ quantum physicists have tried to come up with "quantum versions" of recurrent and feed-forward neural networks. The models were attempts to translate the modular structure as well as the nonlinear activation functions of neural networks into the language of quantum algorithms. However, one could argue that chains of linear and nonlinear computations are rather "unnatural" for quantum computers [#]_.
```
[the remark this answers is not in the document]
Comment (mariaschuld) [reply glossary/quantum_neural_network.rst:21]: I am so guilty of self-citing, but it is the only overview I know...

File: glossary/quantum_neural_network.rst [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+More recent research has tackled this problem, suggesting special measurement schemes or modifications of the neural nets that make them more amenable to quantum computing, but the advantage of these models for machine learning is still not conclusively established.
+
+Quantum versions of Boltzmann machines
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[Boltzmann machines](https://en.wikipedia.org/wiki/Boltzmann_machine), which are probabilistic graphical models that can be understood as stochastic recurrent neural networks, play an important role in the quantum machine learning literature. For example, it was suggested to use samples from a quantum computer to train classical Boltzmann machines, or to interpret spins as physical units of a "quantum" Boltzmann machine model.
```
Comment (mariaschuld) [inline glossary/quantum_neural_network.rst:28]:
```suggestion
[Boltzmann machines](https://en.wikipedia.org/wiki/Boltzmann_machine), which are probabilistic graphical models that can be understood as stochastic recurrent neural networks, play an important role in the quantum machine learning literature. For example, `it was suggested <https://mdenil.com/static/papers/2011-mdenil-quantum_deep_learning-nips_workshop.pdf>`_ to use samples from a quantum computer to train classical Boltzmann machines, or to interpret spins as physical units of a `"quantum" Boltzmann machine model <https://arxiv.org/abs/1601.02036>`_.
```

File: glossary/quantum_neural_network.rst [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+More recent research has tackled this problem, suggesting special measurement schemes or modifications of the neural nets that make them more amenable to quantum computing, but the advantage of these models for machine learning is still not conclusively established.
+
+Quantum versions of Boltzmann machines
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[Boltzmann machines](https://en.wikipedia.org/wiki/Boltzmann_machine), which are probabilistic graphical models that can be understood as stochastic recurrent neural networks, play an important role in the quantum machine learning literature. For example, `it was suggested <https://mdenil.com/static/papers/2011-mdenil-quantum_deep_learning-nips_workshop.pdf>`_ to use samples from a quantum computer to train classical Boltzmann machines, or to interpret spins as physical units of a `"quantum" Boltzmann machine model <https://arxiv.org/abs/1601.02036>`_.
```
Comment (mariaschuld) [inline glossary/quantum_neural_network.rst:28]:
```suggestion
`Boltzmann machines <https://en.wikipedia.org/wiki/Boltzmann_machine>`_, which are probabilistic graphical models that can be understood as stochastic recurrent neural networks, play an important role in the quantum machine learning literature. For example, `it was suggested <https://mdenil.com/static/papers/2011-mdenil-quantum_deep_learning-nips_workshop.pdf>`_ to use samples from a quantum computer to train classical Boltzmann machines, or to interpret spins as physical units of a `"quantum" Boltzmann machine model <https://arxiv.org/abs/1601.02036>`_.
```

File: glossary/quantum_neural_network.rst [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+Variational circuits 
+~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: ../_static/concepts/qnn2.png
+    :align: center
+    :width: 60%
+    :target: javascript:void(0);
+
+Increasingly, the term "quantum neural network" is used to refer to variational or parametrized quantum circuits. While mathematically rather different from the inner workings of neural networks, the analogy highlights the "modular" nature of quantum gates in a circuit, as well as the wide use of tricks from training neural networks used in the optimization of quantum algorithms.
```
Comment (mariaschuld) [inline glossary/quantum_neural_network.rst:39]:
```suggestion
Increasingly, the term "quantum neural network" is used to `refer to variational or parametrized quantum circuits <https://arxiv.org/abs/1802.06002>`_. While mathematically rather different from the inner workings of neural networks, the analogy highlights the "modular" nature of quantum gates in a circuit, as well as the wide use of tricks from training neural networks used in the optimization of quantum algorithms. 
```

**Pull request merged**